src/core/ext/filters/client_channel/lb_policy/pick_first/pick_first.cc

```diff
@@ -418,46 +606,47 @@ void PickFirst::PickFirstSubchannelData::ProcessConnectivityChangeLocked(
   //    select in place of the current one.
   // If the subchannel is READY, use it.
   if (new_state == GRPC_CHANNEL_READY) {
-    subchannel_list()->set_in_transient_failure(false);
+    subchannel_list_->in_transient_failure_ = false;
     ProcessUnselectedReadyLocked();
     return;
   }
   // If we haven't yet seen the initial connectivity state notification
   // for all subchannels, do nothing.
-  if (!subchannel_list()->AllSubchannelsSeenInitialState()) return;
+  if (!subchannel_list_->AllSubchannelsSeenInitialState()) return;
   // If we're still here and this is the initial connectivity state
   // notification for this subchannel, that means it was the last one to
   // see its initial notification.  Start trying to connect, starting
   // with the first subchannel.
   if (!old_state.has_value()) {
-    subchannel_list()->subchannel(0)->ReactToConnectivityStateLocked();
+    subchannel_list_->subchannels_.front().ReactToConnectivityStateLocked();
     return;
   }
   // Ignore any other updates for subchannels we're not currently trying to
   // connect to.
-  if (Index() != subchannel_list()->attempting_index()) return;
+  if (Index() != subchannel_list_->attempting_index_) return;
```

Comment:
Lack of encapsulation for `attempting_index_` seems unusual.

Reply:
I'm not sure what you mean by "encapsulation".  If you mean the fact that we're directly accessing a data member, I think it's fine, because this access is coming from `SubchannelData`, which is nested inside of `SubchannelList` and can therefore access its data members by normal C++ scoping rules.  In effect, `SubchannelData` is part of the implementation of `SubchannelList`, so it's fine to access its private data members.  We use this pattern quite a bit.